src/Veldrid.Utilities/ObjParser.cs

```diff
@@ -499,22 +597,53 @@ public MeshGroup(string name, string material, Face[] faces)
         /// <summary>
         /// An OBJ file construct describing the indices of vertex components.
         /// </summary>
-        public struct FaceVertex
+        public readonly struct FaceVertex : IEquatable<FaceVertex>
```

Comment:
This is a breaking change and needs to be reverted.

Reply:
The `readonly` part can easily be reverted. The `IEquatable<FaceVertex>` removes millions of allocations as the mesh simplifier`s `Dictionary` cast every `FaceVertex` to `object` to check for equality.

Reply:
That's fine, adding `readonly` is the only breaking change here, `IEquatable` can stay.